packages/gatsby/src/services/initialize.ts

```diff
@@ -658,6 +659,10 @@ export async function initialize({
 
   const workerPool = WorkerPool.create()
 
+  if (process.env.GATSBY_GRAPHQL_TYPEGEN) {
```

Comment:
From this part of the code base it's hard to see that this is only ran in develop. Add a comment or make the env var more explicit around develop.